### ISSUE TITLE
Ability to rename channels

### ISSUE THREAD

**Issue by scanlime** [Chocobozzz/PeerTube#1603]
My server has some older channels with UUIDs for names, from before the URL could be customized. Is there a migration path for channels that need to be renamed for any reason?

I'm imagining something where the old name is kept as a redirect, and the new name becomes a canonical default.

-----

For example, this user has channels in both the old style and the new style:

https://diode.zone/accounts/scanlime/video-channels

old style channels have hard-to-remember URLs like:

https://diode.zone/video-channels/ad4dfaf9-8593-447f-b08a-6ba93b3a79ac/videos

whereas a new-style channel is much nicer:

https://diode.zone/video-channels/eyesopod/videos

**Comment by Chocobozzz** [comment]
Related to https://github.com/Chocobozzz/PeerTube/issues/549

**Comment by scanlime** [comment]
I'm not sure if this is best to open a separate issue for, as it seems like perhaps the same underlying root cause as this one, but I've also had users request to change their account username.

**Comment by drzraf** [comment]
When creating a user `<userA>` the default channel seems to be `userA_channel` and I can't find a way to rename these. Does a simple postgresql-like (or similar) command that would allow this?

**Comment by Chocobozzz** [comment]
No, you need to create another channel with the username you want and delete your default channel.

**Comment by tonyppe** [comment]
Hi all, just a question on this. I can understand the bigger picture and doing this may break federation or some component(s) there.  But how about a completely stand alone instance that will never be federated? Can I achieve this today somehow?

Thank you

**Comment by jayrope** [comment]
+1 on account migration

**Comment by Chocobozzz** [comment]
Merging this issue in https://github.com/Chocobozzz/PeerTube/issues/1980
